### PR TITLE
release: sync pre-0.0.1 to main

### PR DESCRIPTION
**Pull request by alexhawat**
## What?

Reviews now get default linters and scanners for A-tier and B-tier languages that used to stay opt-in, plus new coverage for TypeScript, Python security, Go and Rust advisories, unused code, copy-paste detection, and typos.

- Go, Rust, Ruby, and PHP get a default lint path when the repo looks like those languages.
- Nine new tools cover TypeScript, Python security and dead code, Go and Rust advisories, unused JS exports, copy-paste, and typos.
- A-tier languages that had no default signal now have one, including Java, C/C++, Kotlin, Swift, SQL, CSS, HTML, YAML, Make, Markdown, and Terraform.
- B-tier languages get the same treatment: Lua, Fortran, Rego, PowerShell, Batch, Liquid, Smarty, Ember, and Prisma.
- Findings from newly default tools still show up on the PR instead of being dropped when the tool does not speak SARIF.

## Why?

Coverage sweep #309–#339. PRs in those stacks used to ship with an empty default analyzer set unless someone opted in by hand. That work landed on the integration branch via #376 and is still missing from main.

## Note

Includes merged wave branch #376 (open-issues-sweep 2026-08-19e). Issues #309–#339 stay open; several have remaining scope beyond the auto-enable and new-manifest work.

## Test plan

- [ ] `make ci` on a clean tree after merge
- [ ] `make catalog-check`
- [ ] Spot-check a review on a multi-language PR: default analyzers fire, findings survive parse

## Related PR(s)/Issue(s)

#376
Related: #309 #310 #312 #313 #315 #316 #317 #318 #319 #320 #321 #323 #325 #326 #327 #328 #329 #330 #331 #332 #333 #334 #335 #336 #337 #338 #339

### CODE REVIEW

File: src/mergecraft/analyzers/parsers/cargo_audit_json.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    evidence = [f"package: {pkg_name} {pkg_version}".strip()] if pkg_name else []
+    return make_finding(
+        tool=manifest.id,
+        rule_id=rule_id,
+        category=category,
+        severity=map_native_severity(manifest, native_level),
+        confidence=map_confidence(None),
+        message=title,
+        path=path,
+        start_line=1,
```
Comment (sevn-one) [inline src/mergecraft/analyzers/parsers/cargo_audit_json.py]: cargo-audit findings still fake a line-1 anchor: `start_line=1, end_line=1` are hardcoded here. The CHANGELOG explicitly fixed the same class of bug for the sibling repo-scope vuln parsers in this PR ('bundler-audit and cargo-deny findings no longer fake a line-1 GitHub anchor'), and cargo-audit was left behind. Cargo.lock line 1 is the `# This file is automatically @generated` header — typically not in the diff — so inline comments anchor to a non-diff line, and because line=1 is non-None, `budget._is_body_only_finding` keeps these findings competing for inline budget at that misleading anchor rather than the mechanical section. Recommend `start_line=None, end_line=None` (matching `parse_bundler_audit_json` / `parse_cargo_deny_json`).
```suggestion
        start_line=None,
        end_line=None,
```

**Review by sevn-one — COMMENTED** [review]
Approve. Comprehensive, well-tested coverage sweep: default linter/scan auto-enable for A/B-tier languages, 10 new catalog manifests with native parsers, marker-only lint-target skipping, config-presence gates (rubocop/brakeman/shopify/ember/sqlfluff/prisma-lint/phpstan), and the line-optional Finding change threaded through scoping, dedup, budget, SARIF export, and CLI. Logic, security (network allowlists, no new privileged execution), and diff-scoping are sound and pinned by extensive tests. One Minor inconsistency below on cargo-audit's line anchor.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32408595274)

**Comment by sevn-one** [comment]
Reviewed #387 (release: sync pre-0.0.1 to main).

Scope: 10 new catalog manifests (bandit, bundler-audit, cargo-audit, cargo-deny, govulncheck, jscpd, knip, tsc, typos, vulture), auto-enable flips for A/B-tier linters, native non-SARIF parsers, marker-only lint-target skipping (`lint_files`), config-presence gates (D11/D12/D16/D17), and the `Finding.start_line` → optional change threaded through scoping, dedup, budget, baseline suppression, SARIF export, and CLI.

Findings:
- Analyzers pre-merge row: skipped (all catalog analyzers unavailable in this sandbox; no analyzer findings).
- One Minor, non-blocking: `cargo_audit_json._advisory_finding` still hardcodes `start_line=1/end_line=1`, contradicting the PR's own CHANGELOG fix for bundler-audit/cargo-deny — anchors findings at Cargo.lock:1 (autogenerated header, usually not in diff) and keeps them off the body-only/mechanical path. Left an inline comment with a one-line suggestion (`None`).

Verdict: approve.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32408595274)

File: src/mergecraft/analyzers/adapters.py [hunk 2]
```diff
@@ -369,8 +378,24 @@ def run_adapter(
...
+    if not outcome.ran:
+        reason = outcome.output or f"skipped {tool_id}: analyzer did not run"
+        return AdapterRunResult(findings=[], skipped=True, skip_reason=reason)
+
+    if tool_id == "jscpd":
+        report = scratch_dir / "jscpd-report.json"
+        if report.is_file():
+            outcome = replace(outcome, output_path=str(report))
+
+    if outcome.output_path is None:
```
Comment (sevn-one) [inline src/mergecraft/analyzers/adapters.py:398]: _Functional Correctness_ | _Major_ | _Quick win_ | _certain_

A successful analyzer that emits no stdout has `output_path=None`, so this branch classifies clean runs such as `tsc --noEmit` as unavailable instead of passed. Distinguish exit-0 empty output from a run that never executed, and add an adapter regression test for a successful empty-output analyzer.

File: src/mergecraft/analyzers/parsers/tsc_pretty.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+_ANSI_SGR = re.compile(r"\x1b\[[0-9;]*m")
+
+
+def parse_tsc_pretty(raw: str, *, manifest: AnalyzerManifest, repo_root: Path) -> list[Finding]:
+    category = taxonomy_category(manifest)
+    findings: list[Finding] = []
+    for line in raw.splitlines():
+        stripped = _ANSI_SGR.sub("", line).strip()
+        match = _PAREN_DIAG.match(stripped) or _COLON_DIAG.match(stripped)
+        if match is None:
```
Comment (sevn-one) [inline src/mergecraft/analyzers/parsers/tsc_pretty.py]: _Functional Correctness_ | _Major_ | _Quick win_ | _certain_

`tsc` also emits project/config diagnostics without a file location (for example `error TS18003: No inputs were found ...`), and both regexes silently discard them. Because the pipeline treats zero parsed findings as a pass even when `tsc` exited nonzero, parse these as body-only findings or reject nonempty unmatched output, with a regression test.

**Review by sevn-one — CHANGES_REQUESTED** [review]
Requesting changes for two correctness issues in the newly default analyzer path: successful empty-output runs are classified as unavailable, and unlocated TypeScript diagnostics can be silently reported as a pass. The prior `cargo-audit` anchor thread was addressed on the latest commit.

---
*via mergecraft* · `openai/gpt-5.6-sol` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32408595274)

File: src/mergecraft/analyzers/adapters.py (hunk 2 above)
Comment (alexhawat) [reply src/mergecraft/analyzers/adapters.py:398]: Addressed in `14b1b2c5`: a real exit-0 run with no persisted stdout (e.g. clean `tsc --noEmit`) now returns empty findings as a pass instead of `skipped` / unavailable. Added `test_successful_empty_stdout_is_passed_not_unavailable`.

File: src/mergecraft/analyzers/parsers/tsc_pretty.py (hunk 3 above)
Comment (alexhawat) [reply src/mergecraft/analyzers/parsers/tsc_pretty.py]: Addressed in `14b1b2c5`: location-less `tsc` lines such as `error TS18003: ...` now parse as body-only findings (`start_line=None`). Added `test_tsc_project_diagnostic_without_location_is_body_only`.

File: src/mergecraft/analyzers/parsers/cargo_audit_json.py (hunk 1 above)
Comment (alexhawat) [reply src/mergecraft/analyzers/parsers/cargo_audit_json.py]: Already landed in `d82761c3`: cargo-audit advisories now use `start_line=None, end_line=None` (no fake Cargo.lock:1 anchor), with `test_cargo_audit_lockfile_finding_omits_invented_line`.

**Review by sevn-one — CHANGES_REQUESTED** [review]
Requesting changes — CI fails on all 8 Verify jobs due to a single compile-time defect.

## CI failures — underlying defect

All 8 failing jobs (`Verify (static + build)`, `Verify (tests 1/2)`, `Verify (tests 2/2)`, `Verify (integration)`, each ×2) trace to one root cause: **`src/mergecraft/analyzers/parsers/cargo_audit_json.py` passes `end_line` twice to `make_finding` (lines 59–60)**, producing a compile-time `SyntaxError: keyword argument repeated: end_line`.

- `parsers/__init__.py:12` imports this module at package load, so the whole `mergecraft.analyzers.parsers` package is unimportable. The failure propagates through `mergecraft.mcp.server` → `static_checks` → `ci.evidence`, breaking every test job: `313 failed, 1126 passed, 105 errors` (e.g. `test_remaining_format_parsers.py`, `test_auto_enabled_native.py`, `test_sarif.py`) plus collection errors in integration.
- The `static + build` job fails at `ruff check` with `invalid-syntax: Duplicate keyword argument "end_line"` → `cargo_audit_json.py:60:9`.
- mergeCraft's own MCP tools (`create_pull_request_review`, `submit_review_verdict`, `analyze_ci_failures`) also error on the same `SyntaxError` when they re-parse this PR's failing CI, so CI itself could not analyze this run.

**Fix:** delete line 60 (`end_line=1,`). `make_finding` already falls back to `end_line = start_line` when `None` (`finding.py:105–106`), which is exactly the PR's intent for cargo-audit findings that carry no line.

## Prior threads (still unresolved on this head)

- `adapters.py:398` — a clean run with empty stdout (`output_path=None`) is classified as `unavailable`/skipped rather than passed (affects `tsc --noEmit` on a clean tree).
- `tsc_pretty.py:41` — project/config-level `tsc` diagnostics without a file location are silently dropped, so a nonzero `tsc` run can parse to zero findings and read as a pass.

Both are legitimate and should be addressed alongside the blocker. Please also run the test plan items (`make ci`, `make catalog-check`) once the SyntaxError is fixed.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32409621520)

**Comment by sevn-one** [comment]
Reviewed PR #387 (release sync pre-0.0.1 → main) and requested changes.

**Root cause of all 8 failed CI jobs:** `src/mergecraft/analyzers/parsers/cargo_audit_json.py:59-60` passes `end_line` twice to `make_finding` — a compile-time `SyntaxError`. The parsers package import fails (`parsers/__init__.py:12`), cascading to 313 test failures + 105 errors, the `static + build` ruff failure (`invalid-syntax` at line 60), integration collection errors, and even mergeCraft's own MCP tools erroring on the same message. Fix: delete line 60 (`end_line=1`); `make_finding` already defaults `end_line = start_line`.

Also flagged two unresolved prior threads (adapters.py:398 empty-stdout classified as unavailable; tsc_pretty.py:41 dropped unlocated diagnostics). Verifier confirmed the Critical finding; review submitted as CHANGES_REQUESTED at head 54904d5.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32409621520)

**Review by sevn-one — COMMENTED** [review]
Approved. The prior review threads are resolved with targeted regression tests: cargo-audit no longer has the duplicate/fake line anchor, clean empty-output analyzer runs pass, and location-less `tsc` diagnostics surface as body-only findings. The supplied failed suite was for older commit `54904d5`; the current head `14b1b2c` has a successful Verify suite.

---
*via mergecraft* · `openai/gpt-5.6-sol` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32409621520)

**Review by sevn-one — COMMENTED** [review]
Approved. Prior threads are resolved on head `14b1b2c`: cargo-audit no longer fakes a line-1 anchor, clean empty-stdout runs pass, and location-less `tsc` diagnostics surface as body-only findings. The coverage sweep is well-tested and diff-scoped; CI is green on this head.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32410604714)

**Comment by sevn-one** [comment]
Reviewed PR #387 (release: sync pre-0.0.1 to main). All prior review threads are resolved on head `14b1b2c`: cargo-audit's fake line-1 anchor removed, successful empty-stdout runs pass (adapters.py), and location-less `tsc` diagnostics surface as body-only findings (tsc_pretty.py). Verified the newly default analyzer path — marker-only lint-target skipping, config-presence gates, line-optional Finding threading, jscpd scratch-report handling, and diff scoping — plus security posture (no new privileged execution, network allowlists on vuln scanners). No blocking issues. Submitted verdict: **approve**; review published on the PR.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32410604714)

File: src/mergecraft/analyzers/parsers/tsc_pretty.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        if located is not None:
+            start_line: int | None = coerce_line(located.group("line"))
+            path = resolve_repo_relative_path(located.group("path"), repo_root=repo_root)
+            match = located
+        elif bare is not None:
+            start_line = None
+            path = _PROJECT_PATH
+            match = bare
+        else:
+            continue
```
Comment (sevn-one) [inline src/mergecraft/analyzers/parsers/tsc_pretty.py:55]: Non-empty unmatched `tsc` output still becomes a clean pass. This is reachable now that `tsc` auto-enables on any `.ts`/`.tsx` change: without a root `tsconfig.json`, `tsc --noEmit --pretty false` prints help and exits nonzero, none of these regexes match, and the pipeline records zero findings as `passed`. Fail closed on non-empty unmatched output (or propagate the exit status), with an adapter-level regression test.

File: src/mergecraft/analyzers/parsers/tsc_pretty.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        stripped = _ANSI_SGR.sub("", line).strip()
+        located = _PAREN_DIAG.match(stripped) or _COLON_DIAG.match(stripped)
+        bare = None if located is not None else _BARE_DIAG.match(stripped)
+        if located is not None:
+            start_line: int | None = coerce_line(located.group("line"))
+            path = resolve_repo_relative_path(located.group("path"), repo_root=repo_root)
+            match = located
+        elif bare is not None:
+            start_line = None
+            path = _PROJECT_PATH
```
Comment (sevn-one) [inline src/mergecraft/analyzers/parsers/tsc_pretty.py:52]: This parsed body-only diagnostic is still dropped on a `.ts`-only diff: scoping first requires a changed hunk for `finding.path`, but this assigns `tsconfig.json`, so TS18003 disappears unless that config is also changed. Preserve project diagnostics through scoping and add a parser-to-pipeline regression test for a source-only diff.

**Review by sevn-one — CHANGES_REQUESTED** [review]
Requesting changes because `tsc` can still report a clean pass after a failed run: unmatched non-empty output is ignored, and parsed project-level diagnostics are dropped by diff scoping on `.ts`-only changes.

---
*via mergecraft* · `openai/gpt-5.6-sol` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32410604714)

File: src/mergecraft/analyzers/parsers/tsc_pretty.py (hunk 4 above)
Comment (alexhawat) [reply src/mergecraft/analyzers/parsers/tsc_pretty.py:55]: Addressed in `0c2f7a14`: nonempty unmatched `tsc` output (help text, garbage) now raises in `parse_tsc_pretty`, and the adapter treats that as a parse failure rather than a pass. Added `test_tsc_help_stdout_is_parse_failure_not_pass`.

File: src/mergecraft/analyzers/parsers/tsc_pretty.py (hunk 5 above)
Comment (alexhawat) [reply src/mergecraft/analyzers/parsers/tsc_pretty.py:52]: Addressed in `0c2f7a14`: body-only findings (`start_line is None`) now survive `apply_scope_exceptions` / `scope_findings`, so a TS18003 on `tsconfig.json` is kept for a `.ts`-only diff. Added `test_tsc_project_diagnostic_survives_ts_only_diff_scope`.

**Review by sevn-one — COMMENTED** [review]
Approve. The prior requested changes are all resolved on head `0c2f7a1`: cargo-audit no longer emits a fake line-1 anchor (`start_line=None`), clean empty-stdout runs are classified as passed, unmatched nonempty `tsc` stdout fails closed via `require_diagnostic_text`, and project-level `tsc` diagnostics survive `.ts`-only diff scoping through the `start_line is None` keep in `apply_scope_exceptions` — each with targeted regression tests. The broader sweep (auto-enable flips, 10 new manifests with native parsers, marker-only lint-target skipping, config-presence gates for rubocop/brakeman/shopify/ember/sqlfluff/prisma-lint/phpstan) is internally consistent: severity maps match parser native severities, network allowlists are scoped, and the optional-line Finding threading is coherent across scoping, dedup, budget, SARIF export, and CLI. CI is green on this head.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32411981677)

**Comment by sevn-one** [comment]
Reviewed PR #387 (release: sync pre-0.0.1 to main). Verified the current head `0c2f7a1` resolves every concern raised in the last CHANGES_REQUESTED review: tsc fails closed on unmatched nonempty stdout (`require_diagnostic_text`), and project-level tsc diagnostics survive .ts-only diff scoping (start_line-None keep in `apply_scope_exceptions`), each with regression tests; cargo-audit's fake line-1 anchor and the empty-stdout pass handling were already fixed on prior commits. Reviewed the full sweep end-to-end (catalog manifests/severity maps vs parser native severities, marker-only lint-target skip, config-presence gates, jscpd output-file handling, sarif/dedup/budget/baseline optional-line threading) and found no before-merge blockers. CI green on head. Submitted approved:true.

---
*via mergecraft* · `nous/deepseek/deepseek-v4-flash` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32411981677)

**Review by sevn-one — COMMENTED** [review]
Approved. Head `0c2f7a1` resolves the prior threads: cargo-audit findings are body-only without a fake line anchor, clean empty-output analyzer runs pass, and `tsc` now fails closed on unmatched output while preserving project diagnostics through diff scoping. The targeted regression coverage matches each fix.

---
*via mergecraft* · `openai/gpt-5.6-sol` · [run](https://github.com/alexhawat/mergeCraft/actions/runs/32411981677)
